llvm/test/Feature/Repo/fragments.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+"""
+A little utility for dumping the names and digests from a compilation using its ticket file.
+
+For example:
+
+    $ fragments a.o
+    f:9f79e432840075c073e1b70610aabdd0
+    str:e88ce1bd9c14b6c379492240050ae7a8
+    $
+
+Here `a.o` is a ticket file which references a compilation containing the names 'f' and 'str' with the fragment digests
+noted. By default, the tool will dump _all_ of the names from the compilation, but you can request one or more by name:
+
+    $ fragments a.o f
+    f:9f79e432840075c073e1b70610aabdd0
+
+The --digest-only (-d) switch suppresses the symbol name, so:
+
+    $ fragments --digest-only a.o f
+    9f79e432840075c073e1b70610aabdd0
+
+This can be particularly useful to construct a command-line for the pstore-dump --fragment switch:
+
+    $ pstore-dump --fragment=$(fragments --digest-only a.o f) clang.db
+
+The command-line above will dump the contents of the fragment which corresponds to the name 'f' in 'a.o'.
+
+We can go further and dump multiple fragments by adding the --comma (-c) switch:
+
+    $ pstore-dump --fragment=$(fragments -d -c a.o) clang.db
+
+Note: The utility assumes that both `repo-ticket-dump` and `pstore-dump` are on the PATH.
+
+Dependencies: The PyYAML module must be installed.
+"""
+
+import argparse
+import yaml
```

Comment:
When I wrote this little utility I didn't commit it because of this dependency. I’m concerned about requiring anyone wanting to build the prepo repository having to install 3rd-party software that's not needed by a conventional build of LLVM.

If this tool is useful (and I think it is!) then we should rewrite it in C++, commit it next to repo-ticket-dump, and add it to the collection of tools that can be used to explore a repo.

Reply:
Thanks, I will rewrite it in C++.

Reply:
>
> I will rewrite it in C++.

I have created a new PR #115 for code review.